applications/ExternalSolversApplication/external_includes/feast_solver.h

```diff
@@ -135,6 +135,10 @@ class FEASTSolver: public LinearSolver<TSparseSpaceType, TDenseSpaceType,
 
         // don't validate linear_solver_settings here
         mpParam->ValidateAndAssignDefaults(default_params);
+
+        std::string solver_type = mpParam->GetValue("linear_solver_settings")["solver_type"].GetString();
```

Comment:
Because of the multiplication of the main diagonal elements with -1 for the supported DOFS in the eigen_solver_strategy, not all direct solvers coming with MKL are able to solve the system.
Still they are complex direct solvers.
Maybe there is a better place to make such a check.

Reply:
The contract between the two parts of the code should be defined by the interface alone, and any failure to complete that on the side of the solver should be signaled by an exception from that solver. I think it should not be coded here.

Reply:
makes sense, i will try if i can check this.

Reply:
I did not find a solution for this, so i simply removed the ones that where not working, because currently they are not used elsewhere.